City-of-Helsinki/parking-permits
# PV-449: Create audit logger app for structured logging

## Description

Adds a new app, `audit_logger`, which is an app used for creating audit log records as specified in the Jira ticket. Note that this isn't yet used in anything; this is just the system that will be used for creating the logs.

Based on [django-db-logger](https://github.com/CiCiUi/django-db-logger).

## Context

We need to have more structured logging for auditing. This is step one for that; a logger that creates structured log messages and saves them in the database.

[PV-449](https://helsinkisolutionoffice.atlassian.net/browse/PV-449)

## How Has This Been Tested?

Unit tests. Not perfect coverage.

## Manual Testing Instructions for Reviewers

You can do something similar to AuditLoggerTest.test_usage in shell, for example:

```py
from audit_logger import (
    AuditLogHandler,
    AuditMessage,
    AuditType,
    EventType,
    Operation,
    Reason,
    Status,
)

logger = logging.getLogger("audit_logger")
logger.addHandler(AuditLogHandler())
logger.setLevel(logging.DEBUG)
```

...and then log some messages, e.g.

```py
logger.info(
    AuditMessage(
        "Hello world!",
        actor=<some model instance>,
        target=<some model instance>,
        reason=Reason.SELF_SERVICE,
        operation=Operation.CREATE,
        audit_type=AuditType.APP,
        event_type=EventType.APP,
        status=Status.SUCCESS,
        origin="my-origin",
    )
)
```

...and then check the Django admin if there's any new entries.
